lib/ansible/modules/cloud/misc/helm_cli.py

```diff
@@ -0,0 +1,595 @@
+command:
+  type: str
+  description: Full `helm` command built by this module, in case you want to re-run the command outside the module or debug a problem.
+  returned: always
+  sample: helm upgrade ...
+"""
+
+try:
+    import yaml
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+from ansible.module_utils.basic import AnsibleModule
+
+module = None
+is_helm_2 = True
+
+
+# get Helm Version
+def get_helm_client_version(command):
+    is_helm_2_local = True
+    version_command = command + " version --client --short"
+    rc, out, err = module.run_command(version_command)
+
+    if not out.startswith('Client: v2'):
+        is_helm_2_local = False
+        # Fallback on Helm 3
+        version_command = command + " version --short"
+        rc, out, err = module.run_command(version_command)
+
+        if rc != 0:
+            module.fail_json(
+                msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+                command=version_command
+            )
+
+    elif rc != 0:
+        module.fail_json(
+            msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+            command=version_command
+        )
+
+    return is_helm_2_local
+
+
+# Get Values from deployed release
+def get_values(command, release_name, release_namespace):
+    get_command = command + " get values --output=yaml " + release_name
+
+    if not is_helm_2:
+        get_command += " --namespace=" + release_namespace
+
+    rc, out, err = module.run_command(get_command)
+
+    if rc != 0:
+        module.fail_json(
+            msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+            command=get_command
+        )
+
+    # Helm 3 return "null" string when no values are set
+    if not is_helm_2 and out.rstrip("\n") == "null":
+        return yaml.safe_load('{}')
+    else:
+        return yaml.safe_load(out)
+
+
+# Get Release from all deployed releases
+def get_release(state, release_name, release_namespace):
+    if state is not None:
+        if is_helm_2:
+            for release in state['Releases']:
+                # release = {k.lower(): v for k, v in release.items()} # NOT WORKING wit python 2.6
+                release_lower = dict()
+                for k, v in release.items():
+                    release_lower[k.lower()] = v
+                release = release_lower
+                if release['name'] == release_name and release['namespace'] == release_namespace:
+                    return release
+        else:
+            for release in state:
+                if release['name'] == release_name and release['namespace'] == release_namespace:
+                    return release
+    return None
+
+
+# Get Release state from deployed release
+def get_release_status(command, release_name, release_namespace):
+    list_command = command + " list --output=yaml "
```

Comment:
I guess it is a bit late now, but in general, avoid composing commands as strings. Use a list instead with every part of the command in one element. That way, you won't have problems with escaping stuff or having spaces or other crazy things in filenames.